### PR TITLE
find lookup plugins which are installed as roles

### PR DESCRIPTION
**Pull request by lopf**
##### SUMMARY
We install additional custom lookup plugins as roles. In `ansible.cfg` we set `lookup_plugins = roles` in order for the runner to find the lookup plugins which were previously installed via `ansible-galaxy install`. We can then set _ANSIBLE_LOOKUP_PLUGINS_ to include the roles path to find additional plugins.

To make it work, we also had to include the fix proposed in #1632.

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME
- `project_update.yml` to install the roles/plugins into _roles_
- `tasks.py` to set _ANSIBLE_LOOKUP_PLUGINS_

##### AWX VERSION

```
8.0.0/develop
```

##### ADDITIONAL INFORMATION

In the future, additional plugins could probably be installed as collections.

### CODE REVIEW

**Comment by AlanCoding** [comment]
> the lookup plugins which were previously installed via ansible-galaxy install.

I want to understand the related use cases, but I'm doing some extrapolation to imagine what you might be doing here. Your lookup plugin should be a `*.py` file. Where is that relative to the other content in your role?

Since setting `ANSIBLE_LOOKUP_PLUGINS` equal to just "roles" previously worked for you, I'd imagine that it would have to be a top-level file.

**Comment by lopf** [comment]
Here's how we do it: We maintain a lookup plugin (`some.py`) in a repository containing a `meta/main.yml` like a regular role. This allows us to install/distribute this plugin via `ansible-galaxy` (also using tags as `version`). The `requirements.yml` file includes something like this (same as a role):

```
- src: git+https://git.corporate.com/scm/sample/some-lookup.git
  name: company.sample
  version: 0.0.1
```

In the local `ansible.cfg` we specify `lookup_plugins` and `roles_path` to the same directory:

```
[defaults]
lookup_plugins = roles
roles_path = roles
```

With this configuration we install the lookup plugins and roles via galaxy install into `roles`. If we execute playbooks locally, it respects the `lookup_plugins` path and plugins are available. In AWX, the roles, collections and other content is being copied into _/tmp/awx_*/_ during the project update. In our case, it correctly copies the lookup plugins from the roles directory.

Problem: The `lookup_plugins` configuration from the original `ansible.cfg` is not respected in AWX runner, so plugins installed into the `roles` directory can't be found. With this PR, _ANSIBLE_LOOKUP_PLUGINS_ is set to the roles destination path so that lookups are being found.

**Comment by AlanCoding** [comment]
> The lookup_plugins configuration from the original ansible.cfg is not respected in AWX runner

I understand that this doesn't work, but I don't know if I agree with this exact verbiage.

I would rephrase it that your config setting does take effect, it's just that AWX no longer puts your plugin in the place that you were expecting. See: https://github.com/ansible/awx/blob/devel/docs/collections.md

In other places, I have seen users use the hack of referencing `../requirements_roles`. In your case that would be `lookup_plugins=../requirements_roles`. There's a lot to not like about that hack. It would make it super awkward to get the same behavior on the CLI vs. in AWX. But there's also things I don't like about the diff in this PR.

If I was pressed to develop a clean solution for this, it would look like some abandoned ideas from https://github.com/ansible/awx/issues/4675#issuecomment-530891361, basically to re-combine the installed content into the `roles` folder of the temporary directory. But I don't think there would be much appetite for that solution just in terms of priority and effort.

**Comment by lopf** [comment]
It's a pragmatic and working solution for our current project structure but I see that it's not the cleanest fix for everybody. I understand that we're supposed to use Ansible collections starting in 2.9, correct? Until we have everything 2.9 compatible (currently on 2.7), we'll have to use the proposed "workaround".

**Comment by shanemcd** [comment]
@AlanCoding are we going to close this now that 2.9 is out?

**Comment by AlanCoding** [comment]
yes, thanks
